modin/experimental/core/execution/native/implementations/hdk_on_native/__init__.py

```diff
@@ -12,3 +12,101 @@
 # governing permissions and limitations under the License.
 
 """Experimental Modin's functionality related to Native execution engine and optimized for HDK storage format."""
+
+import modin.pandas as pd
+from modin.pandas.utils import from_arrow
+from modin.experimental.core.storage_formats.hdk import DFAlgQueryCompiler
+from modin.experimental.core.execution.native.implementations.hdk_on_native.hdk_worker import (
+    HdkWorker,
+)
+
+
+def hdk_query(query: str, **kwargs) -> pd.DataFrame:
+    """
+    Execute SQL queries on the HDK backend.
+
+    DataFrames are referenced in the query by names and are
+    passed to this function as name=value arguments.
+
+    Here is an example of a query to three data frames:
+
+    ids = [1, 2, 3]
+    first_names = ["James", "Peter", "Claus"]
+    last_names = ["Bond", "Pan", "Santa"]
+    courses_names = ["Mathematics", "Physics", "Geography"]
+    student = pd.DataFrame({"id": ids, "first_name": first_names, "last_name": last_names})
+    course = pd.DataFrame({"id": ids, "name": courses_names})
+    student_course = pd.DataFrame({"student_id": ids, "course_id": [3, 2, 1]})
+    query = '''
+    SELECT
+        student.first_name,
+        student.last_name,
+        course.name
+    FROM student
+    JOIN student_course
+    ON student.id = student_course.student_id
+    JOIN course
+    ON course.id = student_course.course_id
+    ORDER BY
+        last_name
+    '''
+    res = hdk_query(query, student=student, course=course, student_course=student_course)
+    print(res)
+
+    Parameters
+    ----------
+    query : str
+        SQL query to be executed.
+    **kwargs : dict
+        DataFrames referenced by the query.
+
+    Returns
+    -------
+    modin.pandas.DataFrame
+        Execution result.
+    """
+    worker = HdkWorker()
+
+    # Table and column names are mapped to the real names
+    # using the WITH statement.
+    if len(kwargs) > 0:
+        alias = []
+        for name, df in kwargs.items():
+            assert isinstance(df._query_compiler, DFAlgQueryCompiler)
+            mf = df._query_compiler._modin_frame
+            if not mf._has_arrow_table():
+                mf._execute()
+            assert mf._has_arrow_table()
+            part = mf._partitions[0][0]
+            at = part.get()
+
+            if part.frame_id is None:
+                part.frame_id = worker.import_arrow_table(at)
+
+            if len(alias) == 0:
+                alias.append("WITH ")
+                alias.append(name)
+                alias.append(" AS (\n")
+            else:
+                alias.append("\n),\n")
+                alias.append(name)
+                alias.append(" AS (\n")
```

Comment:
The main idea of adding all strings to the list is to avoid temporary strings creation in loop. For better readability, I've replaced multiple appends with a single extend.